presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -257,14 +256,10 @@ public OrcRecordReader(
                 writeValidation);
 
         streamReaders = createStreamReaders(orcDataSource, types, hiveStorageTimeZone, presentColumnsAndTypes.build(), streamReadersSystemMemoryContext);
-        if (includedColumnHandles != null) {
-            for (int columnIdx = 0; columnIdx < streamReaders.length; columnIdx++) {
-                ColumnHandle columnHandle = includedColumnHandles.get(columnIdx);
-                if (columnHandle != null && streamReaders[columnIdx] != null && columnHandle.getReferencedSubfields() != null) {
-                    streamReaders[columnIdx].setReferencedSubfields(columnHandle.getReferencedSubfields(), 0);
-                }
-            }
-        }
+        includedColumnHandles.entrySet().stream()
```

Comment:
I felt like the previous changes were more clear.  Does this fix a bug or is it a pure refactor?

Reply:
It was difficult for me to parse the original version, but looks like both you and Tim prefer loops and if statements. Let me change it back.